File: backend/src/main/java/com/gxs/bank/model/KycDocument.java
package com.gxs.bank.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "kyc_documents")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class KycDocument {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(name = "document_type", nullable = false)
    private DocumentType documentType;

    @Column(name = "document_number", nullable = false)
    private String documentNumber;

    @Column(name = "file_url")
    private String fileUrl;

    @Enumerated(EnumType.STRING)
    @Column(name = "verification_status", nullable = false)
    private VerificationStatus verificationStatus;

    @Column(name = "verified_by")
    private UUID verifiedBy;

    @Column(name = "verified_at")
    private LocalDateTime verifiedAt;

    @Column(name = "rejection_reason")
    private String rejectionReason;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (verificationStatus == null) verificationStatus = VerificationStatus.UPLOADED;
    }

    public enum DocumentType { AADHAAR, PAN, VOTER_ID, DRIVING_LICENSE, PASSPORT }
    public enum VerificationStatus { UPLOADED, VERIFIED, REJECTED }
}

File: backend-python/com/gxs/bank/model/KycDocument.py
from datetime import datetime
from enum import Enum
import uuid

from sqlalchemy import Column, DateTime, Enum as SQLEnum, ForeignKey, String
from sqlalchemy.orm import relationship

from com.gxs.bank.config.database import Base


class DocumentType(str, Enum):
    AADHAAR = "AADHAAR"
    PAN = "PAN"
    VOTER_ID = "VOTER_ID"
    DRIVING_LICENSE = "DRIVING_LICENSE"
    PASSPORT = "PASSPORT"


class VerificationStatus(str, Enum):
    UPLOADED = "UPLOADED"
    VERIFIED = "VERIFIED"
    REJECTED = "REJECTED"


class KycDocument(Base):
    __tablename__ = "kyc_documents"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    userId = Column("user_id", String(36), ForeignKey("users.id"), nullable=False)
    user = relationship("User", lazy="select")

    documentType = Column("document_type", SQLEnum(DocumentType), nullable=False)
    documentNumber = Column("document_number", String(255), nullable=False)
    fileUrl = Column("file_url", String(1024), nullable=True)
    verificationStatus = Column("verification_status", SQLEnum(VerificationStatus), nullable=False, default=VerificationStatus.UPLOADED)
    verifiedBy = Column("verified_by", String(36), nullable=True)
    verifiedAt = Column("verified_at", DateTime, nullable=True)
    rejectionReason = Column("rejection_reason", String(1024), nullable=True)
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)


KycDocument.DocumentType = DocumentType
KycDocument.VerificationStatus = VerificationStatus
